File: libs/perl/test-one.pl
#!/usr/bin/env perl
use Web::Microformats2;

if ($#ARGV != 1) {
    print "Usage: test-one.pl <input_file> <base_url>";
    exit 1;
}

my $file = $ARGV[0];
my $base = $ARGV[1];
my $input = do {
    local $/ = undef;
    open my $fh, "<", $file
        or die "could not open $file: $!";
    <$fh>;
};

my $parser = Web::Microformats2::Parser->new;
my %opts = ( url_context => $base );
my $json = $parser->parse( $input, %opts )->as_json;

print $json;

File: libs/ruby/test-one.rb
#!/usr/bin/env ruby

require 'bundler/setup'
require 'microformats'
require 'json'

def print_usage
  puts 'Usage: test-one <input_file> <base_url>'
end

def process_html(file, baseUrl)
  collection = Microformats.parse(file, base: baseUrl)
  puts JSON.pretty_generate(JSON[collection.to_json.to_s])
end

if ARGV[1].nil?
  print_usage
else
  process_html(ARGV[0], ARGV[1])
end
